### PR TITLE
[FOU-470] ANTLR Code Generation

### PR DESCRIPTION
**Pull request by briantu**
## Summary & Motivation
We want to use ANTLR to define a new grammar for asset selection. This PR includes the files generated by ANTLR and a `README.md` on how to generate them.

## Changelog

> Insert changelog entry or delete this section.

### CODE REVIEW

**Comment by briantu** [comment]
* **#25654** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25654?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @briantu and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/antlr_asset_selection.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from dagster._core.definitions.antlr_asset_selection.AssetSelectionLexer import AssetSelectionLexer
+from dagster._core.definitions.antlr_asset_selection.AssetSelectionParser import (
+    AssetSelectionParser,
+)
+from dagster._core.definitions.antlr_asset_selection.AssetSelectionVisitor import (
+    AssetSelectionVisitor,
+)
+
+
+class AntlrAssetSelection:
```
Comment (benpankow) [inline python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/antlr_asset_selection.py:12]: mark as `@experimental`

File: python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/AssetSelection.g4 [hunk 2]
```diff
@@ -0,0 +1,85 @@
+grammar AssetSelection;
```
Comment (benpankow) [inline python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/AssetSelection.g4:1]: would suggest (1) putting all this generated stuff in a subfolder (2) splitting out the generated code into its own PR (3) clearly marking which files are autogenerated, either with a README in the folder or notes at the top of the files

File: python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/fix_imports.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+def fix_imports(file_path):
```
Comment (benpankow) [inline python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/fix_imports.py:1]: some comments on why this is needed/what it does would be good

File: python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/test_antlr_asset_selection.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import pytest
```
Comment (benpankow) [inline python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/test_antlr_asset_selection.py:1]: test files should live in `dagster_tests`

**Comment by salazarm** [comment]
Can you use https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github to mark these files as generated

File: python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# How to set up ANTLR
```
Comment (benpankow) [inline python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/README.md:1]: Do we want to modify this to describe what ANTLR is used for/how to update this for our use-case?

File: python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+pip install antlr4-python3-runtime
+```
+
+## Generate Python Parser
+
+Under the `python_modules/dagster/dagster/_core/definitions/asset_selection` directory, run
+
+```bash
+$ antlr4 -Dlanguage=Python3 -visitor AssetSelection.g4 -o generated
+```
```
Comment (salazarm) [inline python_modules/dagster/dagster/_core/definitions/antlr_asset_selection/README.md:69]: This can be replaced with the make command right

**Review by benpankow — APPROVED** [review]

**Comment by briantu** [comment]
### Merge activity

* **Nov 14, 3:50 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25654).

**Pull request merged**